src/core_plugins/kibana/public/visualize/listing/listing_table_state.js

```diff
@@ -0,0 +1,66 @@
+import {
+  areAllItemsSelected as areAllItemIdsSelected,
+  toggleItem,
+  toggleAll,
+} from 'ui/saved_object_table/item_selection_actions';
+
+export function getInitialState() {
+  return {
+    items: [],
+    selectedIds: [],
+    currentPage: 0,
+    isFetchingItems: true,
+  };
+}
+
+export function turnToNextPage(state) {
+  return {
+    currentPage: state.currentPage + 1,
+    selectedIds: [],
+  };
+}
+
+export function turnToPreviousPage(state) {
+  return {
+    currentPage: state.currentPage - 1,
+    selectedIds: [],
+  };
+}
+
+export function onToggleItem(state, item) {
+  return {
+    selectedIds: toggleItem(state.selectedIds, item.id),
+  };
+}
+
+export function onToggleAll(state, pager) {
+  const pageOfItems = pager.getItemsOnPage(state.items, state.currentPage);
+  const pageOfItemIds = pageOfItems.map(item => item.id);
+  return {
+    selectedIds: toggleAll(state.selectedIds, pageOfItemIds),
+  };
+}
+
+export function areAllItemsSelected(state, pager) {
```

Comment:
I'm not entirely comfortable with the dependency on Pager that we have in this module, especially in this function. I wonder what it would look like if we only depended on pageOfItems instead.